test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java

```diff
@@ -1125,8 +1125,15 @@ public final void testSyntheticSource() throws IOException {
 
     public void testSyntheticSourceIgnoreMalformedExamples() throws IOException {
         assumeTrue("type doesn't support ignore_malformed", supportsIgnoreMalformed());
-        CheckedConsumer<XContentBuilder, IOException> mapping = syntheticSourceSupport(true).example(1).mapping();
+        // This is weird but we need to call it in order to hit the assumption inside
```

Comment:
Previously this test used mapping from `syntheticSourceSupport` which is wrong since `ExampleMalformedValue` contains a mapping that corresponds to the malformed example. F.e. malformed example can be specific to a date format that is specified in the mapping. I can extract this change in a separate PR if desired.

Reply:
Nit: Skip the "This is weird but" prefix. 

Consider adding an example in the comment to clarify.